LadybugTools_oM/ExecuteCommands/RunWindroseCommand.cs

```diff
@@ -0,0 +1,27 @@
+﻿using BH.oM.Adapter;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+
+namespace BH.oM.LadybugTools
+{
+    [Description("Command that, when executed with the LadybugToolsAdapter, generates a windrose from the epw file using the given parameters.\nOutputs a string file path if the OutputLocation is given, or the base64 string representation of the image if no path is given.")]
+    public class RunWindroseCommand : IExecuteCommand
+    {
+        [Description("The path to an EPW file.")]
+        public virtual string EpwFile { get; set; } = "";
+
+        [Description("The analysis period to use for plotting, default to whole non-leap year.")]
+        public virtual AnalysisPeriod AnalysisPeriod { get; set; } = new AnalysisPeriod();
+
+        [Description("The number of directional bins to plot on the windrose.")]
+        public virtual int NumberOfDirectionBins { get; set; } = 36;
+
+        [Description("A Matplotlib colour map. Corresponds to the 'cmap' parameter of plot methods. See https://matplotlib.org/stable/users/explain/colors/colormaps.html for examples of valid keys. Default of 'viridis'.")]
+        public virtual string ColourMap { get; set; } = "viridis";
```

Comment:
Expecting users to use a string key without messing up the casing has risks from a UX perspective. Use an enum instead and provide the support options as enum options.

Reply:
Let's take out the colour map functionality for now. This is going to need a lot more thought.